coc/coc.py

```diff
@@ -1,154 +1,53 @@
-                current season
-            previous_clan_rank (int): The clan member's prior rank in the clan
-                current season
-            donations (int): The clan member's donations given
-                current season
-            donations_received (int): The clan member's donations received
-                current season
-    """
-
-    def __init__(
-        self, tag, name, role, exp_lvl, league_id, league_name,
-        trophies, vs_trophies, clan_rank, previous_clan_rank,
-        donations, donations_received
-    ):
-        self.tag = tag
-        self.name = name
-        self.role = role
-        self.exp_lvl = exp_lvl
-        self.league_id = league_id
-        self.league_name = league_name
-        self.trophies = trophies
-        self.vs_trophies = vs_trophies
-        self.clan_rank = clan_rank
-        self.previous_clan_rank = previous_clan_rank
-        self.donations = donations
-        self.donations_received = donations_received
-
-
-def get(clan_tag, header):
-    """Takes in the clan's tag and returns a Clan object"""
-
-    clan_json = json_response(clan_tag, header)
-    if clan_json['clanLevel'] < 5:
-        donation_upgrade = 0
-    elif clan_json['clanLevel'] < 10:
-        donation_upgrade = 1
-    else:
-        donation_upgrade = 2
-
-    members = []
-    for member in clan_json['memberList']:
-        if 'versusTrophies' not in clan_json:
-            vs_trophies = 0
+import discord
+from redbot.core import commands, Config
+
+from re import sub
+
+class Coc(commands.Cog):
+
+    def __init__(self, bot):
+        self.bot = bot
+        self.config = Config.get_conf(self, identifier=424242)
+        default_user = {"tag": {}}
+        self.config.register_user(**default_user)
+        self.baseurl = "https://api.clashofclans.com/v1/"
+
+    async def initialize(self):
+        cockey = await self.bot.get_shared_api_tokens("cocapi")
+        token = cockey["api_key"]
+        if token is None:
+            raise ValueError("CoC API key has not been set.")
+        self.headers = {
+            "authorization": f"Bearer {token}",
+            "Accept": "application/json",
+        }
+
+    def apirequest(self, url: str):
+        url = self.baseurl + url
+        try:
+            response = httpx.get(url=url, headers=self.headers, timeout=20)
+            return response.json()
+        except Exception as e:
+            await ctx.send(f"Something went wrong: {e}")
+
+    async def find_member(self, ctx, member: discord.Member):
+        for u in self.config.all_users():
+            if discord.utils.escape_markdown(u.display_name.lower()) == discord.utils.escape_markdown(member.display_name.lower()):
+                await ctx.send(self.config.user(u).tag())
+        await ctx.send("No matches were found.")
+
+    def get(self, ctx, tag: str):
+        """Takes in the clan's tag and returns clan info"""
+
+        tag = "clans/%23" + tag.replace("#", "")
```

Comment:
I don't like this, kinda over-complicated, but that's what you have to work with without a wrapper.